hayeon/programmers-2/삼각-달팽이.js

```diff
@@ -0,0 +1,42 @@
+function solution(n) {
+  const arr = Array.from({ length: n }, (_, index) => Array(index + 1).fill(0));
+  let answer = [];
+  let count = 0;
+  let currentX = -1;
+  let currentY = 0;
+
+  while (n > 0) {
+    for (let i = 0; i < n; i++) {
+      currentX++;
+      count++;
+      arr[currentX][currentY] = count;
+    }
+    for (let i = 0; i < n - 1; i++) {
+      currentY++;
+      count++;
+      arr[currentX][currentY] = count;
+    }
+    for (let i = 0; i < n - 2; i++) {
+      currentX--;
+      currentY--;
+      count++;
+      arr[currentX][currentY] = count;
+    }
+    n -= 3;
+  }
+
+  for (let i = 0; i < arr.length; i++) {
+    answer = [...answer, ...arr[i]];
+  }
+
+  return answer;
```

Comment:
```suggestion
  return arr.reduce((acc, cur) => [...acc, ...cur], []);
```
- answer 변수를 선언하고, 재할당하는 작업은 reduce를 통해 간단하게 해결할 수 있습니다.

Reply:
reduce는 생각도 못했는데 reduce로 다시 풀어보겠습니다!! 👍🏻